presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -1534,4 +1537,61 @@ public void testRedundantDistinctLimitNodeRemoval()
                                 anyTree(
                                         tableScan("orders")))));
     }
+
+    @Test
+    public void testScanNodeInPlanExceedException()
+    {
+        Session enableScanNodeInPlanExceedException = Session.builder(this.getQueryRunner().getDefaultSession())
+                .setSystemProperty(MAX_SCAN_NODE_IN_PLAN, Integer.toString(10))
+                .setSystemProperty(PLAN_NODE_LIMIT_ENABLED, Boolean.toString(true))
```

Comment:
Call it `SCAN_NODE_LIMIT_ENABLED` to be consistent.

Reply:
Changed to `LEAF_NODE_LIMIT_ENABLED` as we include both Value and Scan nodes.